opennlp-tools/src/main/java/opennlp/tools/util/ListHeap.java

```diff
@@ -1,194 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License. You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package opennlp.tools.util;
-
-import java.util.ArrayList;
-import java.util.Comparator;
-import java.util.Iterator;
-import java.util.List;
-
-/**
- * This class implements the heap interface using a {@link java.util.List} as the underlying
- * data structure.  This heap allows values which are equals to be inserted.  The heap will
```

Comment:
The intent here is to replace homegrown Heap, ListHeap with PriorityQueue or a similar data structure.  Talked with @kottmann and we expect the Parses to be unique and hence the question of equal items being inserted will not happen. The other option was to use a LinkedList with the overhead of having to sort the list for each add(parse).  This is still WIP and @kottmann is running regression tests, will have to see what works best amongst - PriorityQueue, SortedSet, LinkedList.